packages/dd-trace/src/exporters/common/request.js

```diff
@@ -1,22 +1,33 @@
 'use strict'
 
+// TODO: Add test with slow or unresponsive agent.
+// TODO: Add telemetry for things like dropped requests, errors, etc.
+
 const http = require('http')
 const https = require('https')
-const log = require('../../log')
 const docker = require('./docker')
 const { storage } = require('../../../../datadog-core')
 
-const httpAgent = new http.Agent({ keepAlive: true })
-const httpsAgent = new https.Agent({ keepAlive: true })
+const keepAlive = true
+const maxTotalSockets = 1
+const maxActiveRequests = 8
+const httpAgent = new http.Agent({ keepAlive, maxTotalSockets })
+const httpsAgent = new https.Agent({ keepAlive, maxTotalSockets })
 const containerId = docker.id()
 
+let activeRequests = 0
+
 function request (data, options, keepAlive, callback) {
   if (!options.headers) {
     options.headers = {}
   }
+
+  // The timeout should be kept low to avoid excessive queueing.
+  const timeout = options.timeout || 2000
```

Comment:
I think this is too low for CI Visibility's agentless

Reply:
2 seconds seems like a pretty long time to me. What scenarios would take longer? The thing with timeouts is that now requests are in sequence instead of in parallel. So effectively the timeout before was 1x15 so 15, and now it's 8x2 so 16.